concrete/controllers/single_page/account/edit_profile.php

```diff
@@ -8,23 +8,22 @@
 use Exception;
 use Concrete\Core\Authentication\AuthenticationType;
 use Concrete\Core\Authentication\AuthenticationTypeFailureException;
-use Loader;
 use User;
 use UserAttributeKey;
 use Localization;
+use Concrete\Core\Http\ResponseFactoryInterface;
 
 class EditProfile extends AccountPageController
 {
     public function view()
     {
         $u = new User();
         $profile = UserInfo::getByID($u->getUserID());
-        if (is_object($profile)) {
-            $this->set('profile', $profile);
-        } else {
-            throw new Exception(t('You must be logged in to access this page.'));
+        if (!is_object($profile)) {
+            return $this->app->make(ResponseFactoryInterface::class)->forbidden(t('You must be logged in to access this page.'));
```

Comment:
We want something with a `ResponseFactoryInterface`, whichever is configured by default (and yes, by default [it points to `ResponseFactory`](https://github.com/concrete5/concrete5/blob/0946f81b56a71f053013c9b2d7fc34cc74c5e3f5/concrete/src/Http/HttpServiceProvider.php#L46))